### PR TITLE
java/client: More specific error for failed connections.

### PR DESCRIPTION
**Pull request by michael-berlin**
java/example: Use try-resource blocks instead of close() statements.

@enisoc

### CODE REVIEW

File: java/example/src/main/java/com/youtube/vitess/example/VitessClientExample.java [hunk 1]
```diff
@@ -38,37 +39,39 @@ public static void main(String[] args) throws Exception {
...
-    RpcClient client = new GrpcClientFactory().create(
-        ctx, new InetSocketAddress(hostAndPort.getHostText(), hostAndPort.getPort()));
-    VTGateConn conn = new VTGateConn(client);
-
-    String keyspace = "test_keyspace";
-    Iterable<String> shards = Arrays.asList("0");
-    Map<String, Object> bindVars =
-        new ImmutableMap.Builder<String, Object>().put("msg", "V is for speed").build();
+    try (RpcClient client = new GrpcClientFactory().create(ctx, addr);
+        VTGateConn conn = new VTGateConn(client);) {
```
Comment (enisoc) [inline java/example/src/main/java/com/youtube/vitess/example/VitessClientExample.java]: Is it recommended style to include the unnecessary trailing semicolon? Examples in general Java docs seem to omit it. Same for the Cursor below.

File: java/grpc-client/src/main/java/com/youtube/vitess/client/grpc/GrpcClient.java [hunk 2]
```diff
@@ -275,8 +275,12 @@ static SQLException convertGrpcError(Throwable e) {
           return new SQLInvalidAuthorizationSpecException(sre.toString(), sre);
         case UNAVAILABLE:
           return new SQLTransientException(sre.toString(), sre);
-        default:
-          return new SQLNonTransientException("gRPC StatusRuntimeException: " + e.toString(), e);
+        default: // Covers e.g. UNKNOWN.
+          String advice = "";
+          if (e.getCause() != null && e.getCause() instanceof java.nio.channels.ClosedChannelException) {
```
Comment (enisoc) [inline java/grpc-client/src/main/java/com/youtube/vitess/client/grpc/GrpcClient.java]: The null check is unnecessary before using instanceof.

**Comment by enisoc** [comment]
LGTM with some nits.

**Comment by michael-berlin** [comment]
Thanks! I've addressed all comments.

**Pull request merged**